### PR TITLE
[DOCS] Updates and clarifications to RigidBody2D

### PR DESCRIPTION
**Pull request by cbscribe**
This one needed some love.

### CODE REVIEW

File: doc/classes/RigidBody2D.xml [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-		This node can use custom force integration, for writing complex physics motion behavior per node.
-		This node can shift state between regular Rigid body, Kinematic, Character or Static.
-		Character mode forbids this node from being rotated.
-		As a warning, don't change RigidBody2D's position every frame or very often. Sporadic changes work fine, but physics runs at a different granularity (fixed hz) than usual rendering (process callback) and maybe even in a separate thread, so changing this from a process loop will yield strange behavior.
+		This is the node that implements simulated 2D physics. You do not control a RigidBody2D directly. Instead you apply forces to it (gravity, impulses, etc.) and the physics simulation calculates the resulting movement based on its mass, friction, and other physical properties.
+		A RigidBody2D has 4 behavior modes (see [member mode]):
+		- [b]Rigid[/b]: The body behaves as a physical object.  It collides with other bodies and responds to forces applied to it. This is the default mode.
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: double spaces after first sentence.

File: doc/classes/RigidBody2D.xml [hunk 2]
```diff
@@ -64,14 +67,14 @@
 			<return type="Vector2">
 			</return>
 			<description>
-				Return the applied force vector.
+				Returns the body's total applied force.
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: extra space before newline

File: doc/classes/RigidBody2D.xml [hunk 3]
```diff
@@ -113,7 +116,7 @@
 			<return type="float">
 			</return>
 			<description>
-				Returns the body's moment of inertia. Automatically computed from associated [class CollisionShape2D]s' mass during physic frames. Inertia is not computed the same frame in which the node was added in. Therefore inertia is not computed during the [code]_ready[/code] function.
+				Returns the body's moment of inertia. Computed automatically from the body's mass and assigned [Shape2D]\s during the physics frame. Note that it will not yet have a value in the [code]_ready()[/code] function.
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: - `[Class]\ s` with a space between `\` and `s`
- should the first and second sentences be merged?

File: doc/classes/RigidBody2D.xml [hunk 4]
```diff
@@ -219,7 +222,7 @@
 			<argument index="0" name="torque" type="float">
 			</argument>
 			<description>
-				Set a constant torque which will be applied to this body.
+				Sets the applied torque.
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: extra space before newline

File: doc/classes/RigidBody2D.xml [hunk 5]
```diff
@@ -378,74 +381,74 @@
...
+			If [code]true[/code] internal force integration is disabled for this body. Aside from collision response, the body will only move as determined by the [method _integrate_forces] function.
 		</member>
 		<member name="friction" type="float" setter="set_friction" getter="get_friction">
-			The body friction, from 0 (frictionless) to 1 (max friction).
+			The body's friction. Values range from [code]0[/code] (frictionless) to [code]1[/code] (maximum friction). Default value: [code]1[/code].
 		</member>
 		<member name="gravity_scale" type="float" setter="set_gravity_scale" getter="get_gravity_scale">
-			This is multiplied by the global 2D gravity setting found in "Project &gt; Project Settings &gt; Physics &gt; 2d" to produce RigidBody2D's gravity. E.g. a value of 1 will be normal gravity, 2 will apply double gravity, and 0.5 will apply half gravity to this object.
+			Multiplies the gravity applied to the body. The body's gravity is calculated from the "Default Gravity" value in "Project &gt; Project Settings &gt; Physics &gt; 2d" and/or any additional gravity vector applied by [Area2D]\s. Default value: [code]1[/code].
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: `[Class]\ s`

File: doc/classes/RigidBody2D.xml [hunk 6]
```diff
@@ -458,8 +461,7 @@
 			<argument index="3" name="local_shape" type="int">
 			</argument>
 			<description>
-				Emitted when a body enters into contact with this one. Contact monitor and contacts reported must be enabled for this to work.
-				This signal not only receives the body that collided with this one, but also its [RID] (body_id), the shape index from the colliding body (body_shape), and the shape index from this body (local_shape) the other body collided with.
+				Emitted when a body enters into contact with this one. Reports colliding shape information.  See [CollisionObject2D] for shape index information. [member contact_monitor] must be [code]true[/code] and [member contacts_reported] greater than [code]0[/code].
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: double spaces after 2nd sentence

File: doc/classes/RigidBody2D.xml [hunk 7]
```diff
@@ -472,37 +474,36 @@
 			<argument index="3" name="local_shape" type="int">
 			</argument>
 			<description>
-				Emitted when a body shape exits contact with this one. Contact monitor and contacts reported must be enabled for this to work.
-				This signal not only receives the body that stopped colliding with this one, but also its [RID] (body_id), the shape index from the colliding body (body_shape), and the shape index from this body (local_shape) the other body stopped colliding with.
+				Emitted when a body shape exits contact with this one. Reports colliding shape information.  See [CollisionObject2D] for shape index information. [member contact_monitor] must be [code]true[/code] and [member contacts_reported] greater than [code]0[/code].
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: double spaces after 2nd sentence

File: doc/classes/RigidBody2D.xml [hunk 8]
```diff
@@ -472,37 +474,36 @@
...
-			Static mode. The body behaves like a [StaticBody2D], and can only move by user code.
+			Static mode. The body behaves like a [StaticBody2D] and does not move.
 		</constant>
 		<constant name="MODE_KINEMATIC" value="3">
-			Kinematic body. The body behaves like a [KinematicBody2D], and can only move by user code.
+			Kinematic mode. The body behaves like a [KinematicBody2D], and must be moved by code.
 		</constant>
 		<constant name="MODE_RIGID" value="0">
-			Rigid body. This is the "natural" state of a rigid body. It is affected by forces, and can move, rotate, and be affected by user code.
+			Rigid mode. The body behaves as a physical object.  It collides with other bodies and responds to forces applied to it. This is the default mode.
```
Comment (Noshyaar) [inline doc/classes/RigidBody2D.xml]: double spaces after 2nd sentence

File: doc/classes/RigidBody2D.xml (hunk 1 above)
Comment (cbscribe) [reply doc/classes/RigidBody2D.xml]: I have the hardest time breaking that habit.

File: doc/classes/RigidBody2D.xml (hunk 5 above)
Comment (akien-mga) [reply doc/classes/RigidBody2D.xml]: What's with this weird construct though? It should be `[Class]s` and both DocData and makerst should be fixed to handle it properly if they don't.

Comment (Noshyaar) [reply doc/classes/RigidBody2D.xml]: [Class] converts to `:ref:'Class <Class>'`

From: http://www.sphinx-doc.org/en/stable/rest.html
>it must be separated from surrounding text by non-word characters. Use a backslash escaped space to work around that: thisis\ \*one\*\ word.

Last time I check:
`:ref:'Class <Class>' s` became `Class s`.
`:ref:'Class <Class>'s` was broken.
`:ref:'Class <Class>'\ s` became `Classs`.

Comment (Noshyaar) [reply doc/classes/RigidBody2D.xml]: Without escaping:
image

With escaping:
image

Comment (akien-mga) [reply doc/classes/RigidBody2D.xml]: Yeah, but that's why we should handle that in `makerst.py` and not in the XML itself. It's easy enough to detect `[.*][a-zA-Z0-9]` in a string and output the necessary escape accordingly.

**Pull request merged**